reason-react-tutorial.md

```diff
@@ -0,0 +1,681 @@
+
+Kinda reminds me of this Kyle Simpson quote from "[You Don't Know JS: Async & Performance](https://github.com/getify/You-Dont-Know-JS/blob/master/async%20%26%20performance/README.md)"
+
+> It's impossible to effectively use any abstraction if you don't understand what it's abstracting, and why.
+
+The other abstraction that's important to know about is Reason itself, since ReasonReact is *React in Reason*.  If you're just getting started with Reason I'd recommend having the [Reason docs](https://reasonml.github.io/docs/en/what-and-why) up as you read this article, just in case you need any refreshers on the syntax or language mechanics. I also have an [introductory article](https://blog.logrocket.com/what-makes-reasonml-so-great-c2c2fc215ccb) to Reason that may be a good read if you're just starting out with Reason and want to know what all the hype is about.
+
+In addition, there's one disclaimer I'd like to add. We'll definitely see some "rough edges" and not-quite-finished language features as we go through this tutorial. This is largely because Reason is a newer community, even though it's based on OCaml's more mature community. It's important to remember that the teams behind Reason and ReasonReact are aware of the pain points and awkardness around certain parts of the language, and are working hard to fix them. While certain parts (like async/await or CSS) aren't fully solved yet, there's a lot of really smart people working to solve them right now.
+
+## Getting Started
+
+For this tutorial, we'll be building a Github Search app. This app will have a search bar at the top that takes any topic your heart desires. Then, when the search entry is submitted, we'll query the GitHub REST API and display a list of repositories matching that topic.
+
+Here's a [live link](https://benjamminj.github.io/reason-react-tutorial/) to the app if you want to poke around a little bit. 
+
+I find it helpful to build an app of this size whenever I'm learning something new&mdash;in fact, I often use _this exact app requirements_ to learn a new technology or framework. Solving a familiar problem is a good way to get a feel for an unfamiliar technology. Similar to the way people make to-do lists or Hacker News clones, a GitHub search app is just complex enough that we'll have to do things like state-management and API calls, yet simple enough to build in a single tutorial.
+
+If you're looking to look through the source code you can check out the repo [here](https://github.com/benjamminj/reason-react-tutorial). To code along check out the `getting-started` branch. This will only contain the boilerplate to get a "hello world" on the screen, and then we'll fill in the rest of the app from there.
```

Comment:
I like the idea of providing a `getting-started` branch! Need to do it for my own tutorials where I always provided a separate repository.